src/indexer/merger.rs

```diff
@@ -103,204 +146,387 @@ impl IndexMerger {
     }
 
     fn write_fieldnorms(&self, fast_field_serializer: &mut FastFieldSerializer) -> Result<()> {
-        let fieldnorm_fastfields: Vec<Field> = self.schema
-            .fields()
-            .iter()
-            .enumerate()
-            .filter(|&(_, field_entry)| field_entry.is_indexed())
-            .map(|(field_id, _)| Field(field_id as u32))
-            .collect();
-        self.generic_write_fast_field(
-            fieldnorm_fastfields,
-            &extract_fieldnorm_reader,
-            fast_field_serializer,
-        )
+        for (field_id, field_entry) in self.schema.fields().iter().enumerate() {
```

Comment:
Is there any reason for this to contain field_ids instead of `Field`s?

Reply:
Right now you need to enumerate and create the Field object on your own. (That kind of sucks.)